### PR TITLE
add rake task

### LINKED ISSUE #1050: make sitemap
task for sitemap generation, will include all public asset uris
also set up schedule to keep sitemap up to date

### PR DESCRIPTION
**Pull request by lsat12357**
fixes #1050 
basic rake task to grab all of the assets/sets urls and print xml.
also an update task, grabs new assets/sets and appends. method of appending might be a little wonky though.

### CODE REVIEW

**Comment by jechols** [comment]
I think we should mark set landing pages as being higher priority than individual asset pages so that search engines are more likely to hit those first.  We also need to get pages in the map that are important but aren't in the database.

If I had to guess, I'd say the home page should probably have a priority of 1.0, the list of sets should be 0.9, the set landing pages 0.75, and finally assets should be 0.5 (that's the default, though, so it doesn't have to be specified).  That's my opinion, though - maybe we should get input from stakeholders.

**Comment by lsat12357** [comment]
ok went ahead and added priorities, can tweak further. Also realizing that updater method doesn't handle deletes.  Also jechols notes that if for some reason the updater doesn't get run, someone has to manually figure out what date range to run.

File: lib/tasks/sitemap_gen.rake [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+desc 'Create a sitemap for OregonDigital'
+
+def gen_map_start(f)
+    home_priority = "1.0"
+    setlist_priority = ".9"
```
Comment (jechols) [inline lib/tasks/sitemap_gen.rake]: I'm not certain, but I think this needs to be "0.9", not ".9".  The examples at sitemaps.org always have the leading 0.

File: lib/tasks/sitemap_gen.rake [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    f.puts '<?xml version="1.0" encoding="UTF-8"?>'
+    f.puts '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">'
+    f.puts "<url><loc>http://oregondigital.org/catalog</loc><priority>#{home_priority}</priority></url>"
+    f.puts "<url><loc>http://oregondigital.org/sets</loc><priority>#{setlist_priority}</priority></url>"
+end
+
+def gen_map_body(f, args={})
+  setlanding_priority = ".75"
```
Comment (jechols) [inline lib/tasks/sitemap_gen.rake]: Same issue here - 0.75 instead of .75

**Pull request merged**